ibm/service/kubernetes/resource_ibm_container_worker_pool.go

```diff
@@ -275,7 +275,7 @@ func resourceIBMContainerWorkerPoolCreate(d *schema.ResourceData, meta interface
 
 	d.SetId(fmt.Sprintf("%s/%s", clusterNameorID, res.ID))
 
-	return resourceIBMContainerWorkerPoolRead(d, meta)
+	return resourceIBMContainerWorkerPoolUpdate(d, meta)
 }
```

Comment:
In update method resourceIBMContainerWorkerPoolUpdate we do have other two  d.HasChange("size_per_zone") and d.HasChange("labels") will also diff even if we used them during of create worker pool . Can you add this condition also 

```
if d.HasChange("size_per_zone") && !d.IsNewResource()
```
for both else it will unneceessaty update calls.

Reply:
makes sense, thx. added the check in https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4377/commits/a24101454e59a133560790ea33317c9396d56c41